### PR TITLE
fix: home view performance

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description

**Context:** https://artsy.slack.com/archives/C05EQL4R5N0/p1733916954506419
TLDR;
In this PR, I worked on fixing the home view screen performance and make it more snappy. The time went from 300-500ms to 50-100ms.

**What led to this?**
- many more new home view components
- no memoisation in place
- we rolled back some virtualised lists improvements in this [PR](https://github.com/artsy/eigen/pull/11207) - which led to fixing some bugs on Android and iOS getting slower

**Bonus work:**
Follow react-navigation [optimisation tips](https://reactnavigation.org/docs/6.x/screen#getcomponent) for the main tabs. 
Screenshot 2024-12-11 at 17 47 41

| Before | After |
|--------|--------|
| <Video src="https://github.com/user-attachments/assets/0b8ecb83-0414-4a03-b2ef-251869317d16" /> | <Video src="https://github.com/user-attachments/assets/35e19fa7-ee8f-454e-9500-ea25d2a95c28" /> | 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- improve home view tab performance - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (improve home view tab performance - mounir - MounirDhahri)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 051c81993ce3d9f8674bed06b085fe4d89c85b62
</p>

**Review by damassi — DISMISSED** [review]
Nice, so we can opt into performance improvements by just using `memo`, without needing to be specific with comparator functions. Did not realize this 🎉

**Comment by MounirDhahri** [comment]
Yes, `memo` is the most "lazy" way to skip component re-renders. A lot of the time, it's all you need. Actually, so many design system libraries, wrap each component with a memo. It's something we could have done as well but maybe we can wait for the new compiler and get that out of the box.

**Review by olerichter00 — DISMISSED** [review]
Looks good! 🌟 This memo trick seems to be pretty easy. Are there any downsides, caveats, or situations where using `memo` isn't a good idea?

File: src/app/Components/ArtworkRail/ArtworkRail.tsx [hunk 1]
```diff
@@ -40,74 +41,84 @@ export interface ArtworkRailProps extends ArtworkActionTrackingProps {
...
+              hideCuratorsPickSignal={hideCuratorsPickSignal}
+              {...otherProps}
+            />
+          </Box>
+        )
+      },
+      [hideArtistName, onPress, showPartnerName]
+    )
+
+    const Wrapper =
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Components/ArtworkRail/ArtworkRail.tsx:89]: of course

**Comment by olerichter00** [comment]
Thanks for the explanation, @MounirDhahri! 🙏

**Review by dariakoko — DISMISSED** [review]
Thanks for the improvements Mounir!

File: src/app/Scenes/HomeView/HomeView.tsx [hunk 2]
```diff
@@ -180,12 +180,14 @@ export const HomeView: React.FC = () => {
           refreshControl={<RefreshControl refreshing={isRefreshing} onRefresh={handleRefresh} />}
           onEndReachedThreshold={2}
           stickyHeaderIndices={[0]}
+          initialNumToRender={NUMBER_OF_SECTIONS_TO_LOAD}
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/HomeView/HomeView.tsx:183]: yes it is probably enough in this context but I want to also make it explicit for the `Flatlist` that we don't want to render more than that in the initial batch.

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/HomeView/HomeView.tsx:183]: I think it won't have any impact here so it's fine to remove it

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/HomeView/HomeView.tsx:183]: I will do that

File: src/app/Scenes/Activity/Activity.tests.tsx [hunk 3]
```diff
@@ -16,7 +16,9 @@ describe("ActivityScreen", () => {
       resolveHeaderLastOperation({})
     })
 
-    await waitForElementToBeRemoved(() => screen.queryByTestId("activity-content-placeholder"))
+    await waitForElementToBeRemoved(() => screen.queryByTestId("activity-content-placeholder"), {
+      timeout: 15000,
+    })
```
Comment (dariakoko) [inline src/app/Scenes/Activity/Activity.tests.tsx:21]: Instead of timeout we can use
`await waitForElementToBeRemoved(() => screen.queryByTestId("activity-content-placeholder"))`

**Review by dariakoko — APPROVED** [review]

File: src/app/Scenes/Activity/Activity.tests.tsx (hunk 3 above)
Comment (MounirDhahri) [reply src/app/Scenes/Activity/Activity.tests.tsx:21]: that didn't work actually - it's pretty unstable since any months
Screenshot 2024-12-12 at 15 05 49

Comment (dariakoko) [reply src/app/Scenes/Activity/Activity.tests.tsx:21]: Ah, interesting..

Comment (MounirDhahri) [reply src/app/Scenes/Activity/Activity.tests.tsx:21]: that helped - all next PRs succeeded 
Screenshot 2024-12-12 at 16 34 36
